tests/app/dao/test_jobs_dao.py

```diff
@@ -238,23 +238,24 @@ def test_get_future_scheduled_job_gets_a_job_yet_to_send(sample_scheduled_job):
 
 
 @freeze_time('2016-10-31 10:00:00')
-def test_should_get_jobs_seven_days_old(sample_template):
+def test_should_get_jobs_older_than_retention(sample_template):
     """
-    Jobs older than seven days are deleted, but only two day's worth (two-day window)
+    Jobs older than retention period are deleted, but only two day's worth (two-day window)
```

Comment:
[question] what's this "only two day's worth" mean? We're not deleting things that are older than 3 days if they somehow sneak through?

Reply:
I'm...not sure and I couldn't really tell from reviewing the code. This comment is one of the reasons that I want to do some extra manual testing.

Reply:
The two-days worth comment should be deleted as it's a red-herring. The thing preventing the deletion of the older jobs here is `archived=True`. We should fully understand and document when a job is archived (and why) since that seems to prevent removing the record even once it is old enough that it should be removed. I'm fine with doing that documentation in this PR, but I think I'd prefer to just see a new issue opened (and put into the backlog) and linked to here for context.

Reply:
Ah yes, thank you for pointing that out! Here's an issue for figuring out more: https://github.com/GSA/notifications-api/issues/200